### PR TITLE
UDP commands for MAX! Cube

### PR DESCRIPTION
**Pull request by marcelrv**
* New command class for sending udp commands to MAX! Cube
* Trigger Cube reset from HabminUI tools menu

Signed-off-by: Marcel Verpaalen <marcel@verpaalen.com>

### CODE REVIEW

**Comment by marcelrv** [comment]
looks like the jenkins issue is resolved :)

File: addons/binding/org.openhab.binding.max/src/main/java/org/openhab/binding/max/internal/handler/MaxCubeBridgeHandler.java [hunk 1]
```diff
@@ -193,6 +197,35 @@ public void initialize() {
...
+                            maxConfiguration.serialNumber);
+                    reset.setIpAddress(maxConfiguration.ipAddress);
+                    reset.send();
+                }
+            }
+
+            configuration.put(configurationParameter.getKey(), configurationParameter.getValue());
+        }
+
+        // Persist changes
```
Comment (kaikreuzer) [inline addons/binding/org.openhab.binding.max/src/main/java/org/openhab/binding/max/internal/handler/MaxCubeBridgeHandler.java]: The BaseThingHandler does it this way:
```
            // persist new configuration and notify Thing Manager
            updateConfiguration(configuration);
            callback.configurationUpdated(getThing());
```
The handleConfigurationUpdate is specifically introduced to NOT dispose/initialize the complete handler - so if you can avoid it, you should do so!

File: addons/binding/org.openhab.binding.max/ESH-INF/thing/bridge.xml [hunk 2]
```diff
@@ -34,6 +34,12 @@
 				<description>Binding settings for this device</description>
 				<advanced>false</advanced>
 			</parameter-group>
+			<!-- Trigger actions from habmin -->
+            <parameter-group name="actions">
```
Comment (kaikreuzer) [inline addons/binding/org.openhab.binding.max/ESH-INF/thing/bridge.xml]: you now have tabs/spaces mixed in this file. Could you reapply the formatter here?

File: addons/binding/org.openhab.binding.max/ESH-INF/thing/bridge.xml [hunk 3]
```diff
@@ -34,6 +34,12 @@
...
+                <context></context>
```
Comment (kaikreuzer) [inline addons/binding/org.openhab.binding.max/ESH-INF/thing/bridge.xml]: please remove the line if empty.

File: addons/binding/org.openhab.binding.max/ESH-INF/thing/bridge.xml [hunk 4]
```diff
@@ -34,6 +34,12 @@
...
+                <label>Actions</label>
+                <description></description>
```
Comment (kaikreuzer) [inline addons/binding/org.openhab.binding.max/ESH-INF/thing/bridge.xml]: could you add a description here?

**Comment by marcelrv** [comment]
Hi @kaikreuzer thanks for your review.
I believe I fixed all issues from your comments

**Comment by kaikreuzer** [comment]
Thanks!

**Pull request merged**